### ISSUE TITLE
The built in validators(phone, email etc.) are validating optional parameters

### ISSUE THREAD

**Issue by csiszarattila** [Luracast/Restler#531]
Testcase to reproduce the error:

``` php
/**
 * @param string $name
 * @param string $phone 
*/
public function post($name, $phone = "") 
{
...
}
```

Now if I post phone parameter as an empty string eg.:

``` json
{
  "name": "John Doe",
  "phone": "",
}
```

The phone validator still try to validate the value, and obviously throws an error:

```
Bad Request: Invalid value specified for `phone`. Expecting phone number, a numeric value with optional `+` prefix
```

**Comment by letsbyteit** [comment]
We've encountered this issue, too. A fix would be great.

**Comment by csiszarattila** [comment]
@letsbyteit a quick workaround: use mixed on annotation instead of string, eg.:

```php
/**
* @var mixed $phone
...

**Comment by letsbyteit** [comment]
@csiszarattila As you might have seen, we fixed this issue: https://github.com/Luracast/Restler/pull/533

**Comment by Arul-** [comment]
Fixed in cad72d0cb364951b2ae85f066494ddc4f17e52c0
